src/pages/index.tsx

```diff
@@ -1,10 +1,57 @@
 import Head from 'next/head';
 import { Inter } from '@next/font/google';
 import styles from '@/styles/Home.module.css';
+import { useEffect, useState } from 'react';
+import axios from 'axios';
 
 const inter = Inter({ subsets: ['latin'] });
 
+type GitHubRepository = {
+  description: string;
+  language: string;
+  name: string;
+  stars: number;
+  url: string;
+}
+
+type GitHubUserProfile = {
+  avatarUrl: string;
+  url: string;
+  bio: string;
+  name: string;
+}
+
+type GitHubProfile = {
+  languagesUsed: Map<string, number>;
+  profile?: GitHubUserProfile;
+  topFiveStarredRepos: GitHubRepository[];
+}
+
+type Profile = {
+  gitHub?: GitHubProfile;
+}
+
 export default function Home() {
+
+  const [profile, setProfile] = useState<Profile>({})
+  const [languages, setLanguages] = useState<string[]>([])
+
+  useEffect(()=>{
+    axios.get(`https://ferrata.builtwithdark.com/open-profile/api/v1/profile/ferrata`)   //gotta make the username part dynamic `${username}`
```

Comment:
If we could get a username from the URL path here, we could merge this PR to `main`. I think.

Reply:
yes